src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -923,6 +923,12 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
                     // If this was a ref extension method, receiverArgument must be checked for L-value constraints.
                     // This helper method will also replace it with a BoundBadExpression if it was invalid.
                     receiverArgument = CheckValue(receiverArgument, BindValueKind.RefOrOut, diagnostics);
+
+                    CheckFeatureAvailability(receiverArgument.Syntax, MessageID.IDS_FeatureRefExtensionMethods, diagnostics);
```

Comment:
>CheckFeatureAvailability(receiverArgument.Syntax, MessageID.IDS_FeatureRefExtensionMethods, diagnostics); [](start = 20, length = 105)

I would recommend to perform this check only if method isn't defined in this source module. Helps suppressing cascading diagnostics.